storage/remote/queue_manager.go

```diff
@@ -415,17 +415,19 @@ func NewQueueManager(
 		reshardChan: make(chan int),
 		quit:        make(chan struct{}),
 
-		dataIn:          samplesIn,
-		dataDropped:     newEWMARate(ewmaWeight, shardUpdateDuration),
-		dataOut:         newEWMARate(ewmaWeight, shardUpdateDuration),
-		dataOutDuration: newEWMARate(ewmaWeight, shardUpdateDuration),
+		DataIn:          samplesIn,
+		DataDropped:     NewEWMARate(ewmaWeight, shardUpdateDuration),
+		DataOut:         NewEWMARate(ewmaWeight, shardUpdateDuration),
+		DataOutDuration: NewEWMARate(ewmaWeight, shardUpdateDuration),
 
 		metrics:              metrics,
 		interner:             interner,
 		highestRecvTimestamp: highestRecvTimestamp,
 	}
 
-	t.watcher = wal.NewWatcher(watcherMetrics, readerMetrics, logger, client.Name(), t, walDir, enableExemplarRemoteWrite)
+	if walDir != "" {
```

Comment:
Change not covered by commit description, seems unrelated to PR title.
Please put in a different PR with explanation.